Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -76,11 +81,49 @@ class DataPipelineImplementation: DataPipelineInstance {
     var profileAttributes: [String: Any] {
         get { analytics.traits() ?? [:] }
         set {
-            let userId = analytics.userId ?? analytics.anonymousId
-            analytics.identify(userId: userId, traits: newValue)
+            let userId = analytics.userId
+            guard let userId = userId else {
+                logger.error("No user identified. If you don't have a userId but want to record traits, please pass traits using identify(body: Codable)")
+                return
+            }
+            commonIdentifyProfile(userId: userId, attributesDict: newValue)
         }
     }
 
+    private func commonIdentifyProfile(userId: String, attributesDict: [String: Any]? = nil, attributesCodable: Codable? = nil) {
+        let currentlyIdentifiedProfile = analytics.userId
+        let isChangingIdentifiedProfile = currentlyIdentifiedProfile != nil && currentlyIdentifiedProfile != userId
+        let isFirstTimeIdentifying = currentlyIdentifiedProfile == nil
+
+        if isFirstTimeIdentifying || isChangingIdentifiedProfile {
+            // logger.debug("running hooks profile identified \(userId)")
+            // FIXME: [CDP] Request Journeys to invoke profile identify hooks
+            // hooks.profileIdentifyHooks.forEach { hook in
+            //     hook.profileIdentified(identifier: userId)
+            // }
```

Comment:
I *think* that this profile hook was only intended for having the messagingpush module register an existing device token to a profile. Now that registering devices tokens exists in the tracking module, we may not need a hook/journeys call at all.